File: ScriptGraph/PinToGraph/PropertyScript/JS/ScriptComponentProperty.js
const Amaz = effect.Amaz;

class ScriptComponentProperty {
  constructor() {}

  setProperty(objects, property, value) {
    if (!Array.isArray(objects) || objects.length === 0) {
      return null;
    }
    const jsObject = objects[0].getScript().ref;
    jsObject[property] = value;
  }

  getProperty(objects, property) {
    if (!Array.isArray(objects) || objects.length === 0) {
      return null;
    }
    const jsObject = objects[0].getScript().ref;
    return jsObject[property];
  }
}

exports.ScriptComponentProperty = ScriptComponentProperty;

File: ScriptGraph/PinToGraph/PropertyScript/Lua/ScriptComponentProperty.lua
local ScriptComponentProperty = ScriptComponentProperty or {}
ScriptComponentProperty.__index = ScriptComponentProperty

function ScriptComponentProperty.new()
  local self = setmetatable({}, ScriptComponentProperty)
  return self
end

function ScriptComponentProperty:setProperty(objects, property, value)
  if #objects == 0 then
    return
  end
  local luaObject = Amaz.ScriptUtils.getLuaObj(objects[1]:getScript())
  luaObject[property] = value
end

function ScriptComponentProperty:getProperty(objects, property)
  if #objects == 0 then
    return nil
  end
  local luaObject = Amaz.ScriptUtils.getLuaObj(objects[1]:getScript())
  return luaObject[property]
end

return ScriptComponentProperty